lib/surface/components/dynamic/live_component.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Surface.Components.Dynamic.LiveComponent do
+  @moduledoc """
+  A built-in component that allows users to inject dynamic live components into
+  a Surface template.
+  """
+
+  use Surface.LiveComponent
+
+  alias Surface.AST
+
+  @doc """
+  The module of the component
+  """
+  prop module, :module, required: true
+
+  @doc """
+  The default slot
+  """
+  slot default
+
+  def transform(node) do
+    %AST.Component{props: props, directives: directives, templates: templates, meta: meta} = node
+
+    {%{module: mod}, other_props} = AST.pop_attributes_values_as_map(props, [:module])
+
+    %AST.Component{
+      module: mod,
+      type: :dynamic_live,
+      props: other_props,
+      directives: directives,
+      templates: templates,
+      meta: meta
+    }
+  end
+
+  def render(assigns),
+    do: ~F"""
+      <div/>
```

Comment:
Should we add an empty `div` because it is a live component?

Reply:
yes, without adding an empty html tag the compiler throws an error as a live component must have a root tag. I don't know if an empty div is the right choice, I'm open to any suggestions

Reply:
good point , changed